File: LeetCode Problems/901 to 1000/958. Check Completeness of a Binary Tree/958. Check Completeness of a Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int countNodes(TreeNode root) {
        if (root == null) return 0;
        return countNodes(root.left) + countNodes(root.right) + 1;
    }
    private boolean checker(TreeNode root, int currNode, int totalNode) {
        if (root == null) return true;
        if (currNode >= totalNode) return false;
        return checker(root.left, 2 * currNode + 1, totalNode) && checker(root.right, 2 * currNode + 2, totalNode);
    }
    public boolean isCompleteTree(TreeNode root) {
        return checker(root, 0, countNodes(root));
    }
}

File: LeetCode Problems/901 to 1000/958. Check Completeness of a Binary Tree/958. Check Completeness of a Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isCompleteTree(TreeNode *root) {
        return checker(root, 0, countNodes(root));
    }

private:
    int countNodes(TreeNode *root) {
        if (root == NULL) return 0;
        return countNodes(root->left) + countNodes(root->right) + 1;
    }
    bool checker(TreeNode *root, int currNode, int totalNode) {
        if (root == NULL) return true;
        if (currNode >= totalNode) return false;
        return checker(root->left, 2 * currNode + 1, totalNode) &&
               checker(root->right, 2 * currNode + 2, totalNode);
    }
};